pkg/wasm/utils.go

```diff
@@ -144,9 +144,14 @@ func PredicatesFromHTTPRouteMatch(match gatewayapiv1.HTTPRouteMatch) []string {
 		predicates = append(predicates, predicateFromHeader(headerMatch))
 	}
 
-	// TODO(eguzki): query params. Investigate integration with wasm regarding Envoy params
-	// from https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes
-	// request.query -> string : The query portion of the URL in the format of “name1=value1&name2=value2”.
+	// query param, only consider the first in case of repetition, as per spec
+	queryParams := make(map[gatewayapiv1.HTTPHeaderName]bool)
```

Comment:
Dunno if that's idiomatic Go...

Reply:
its fine, I think you would find most use `queryParams := map[gatewaypiv1.HTTPHeaderName]bool{}` but I don't think it is that important